src/midi/midi_fluidsynth.cpp

```diff
@@ -384,17 +424,25 @@ bool MidiHandlerFluidsynth::Open([[maybe_unused]] const char *conf)
 #endif
 
 	if (chorus_enabled)
-		LOG_MSG("MIDI: Chorus enabled with %d voices at level %.2f, %.2f Hz speed, %.2f depth, and %s-wave modulation",
-		        chorus_voice_count, chorus_level, chorus_speed, chorus_depth,
-		        chorus_mod_wave == fluid_chorus_mod::FLUID_CHORUS_MOD_SINE ? "sine" : "triangle");
+		LOG_MSG("FSYNTH: Chorus enabled with %d voices at level %.2f, %.2f Hz speed, %.2f depth, and %s-wave modulation",
+		        chorus_voice_count,
+		        chorus_level,
+		        chorus_speed,
+		        chorus_depth,
+		        chorus_mod_wave == fluid_chorus_mod::FLUID_CHORUS_MOD_SINE
+		                ? "sine"
+		                : "triangle");
 
 	if (reverb_enabled)
-		LOG_MSG("MIDI: Reverb enabled with a %.2f room size, %.2f damping, %.2f width, and level %.2f",
-		        reverb_room_size, reverb_damping, reverb_width, reverb_level);
+		LOG_MSG("FSYNTH: Reverb enabled with a %.2f room size, %.2f damping, %.2f width, and level %.2f",
+		        reverb_room_size,
+		        reverb_damping,
+		        reverb_width,
+		        reverb_level);
 
 	settings = std::move(fluid_settings);
 	synth = std::move(fluid_synth);
-	channel = std::move(mixer_channel);
+	channel = mixer_channel;
```

Comment:
Suggest keeping the `std::move`, but relaxing the local `const auto mixer_channel = ` to a non-const further up.

The non-const'ing will let the local smart pointer release its holding into `channel` instead of copying by value.

Thanks for cleaning this up - probably been like this a while (where `move` wasn't able to take effect).

Letting compound local rvalue's be `std::move`'d (or return-value-optimized automatically by the compiler when returning them), is the one place to avoid using `const` on them.

Reply:
Cheers, I wasn't completely sure about this, to be honest, just tried to follow what PVS Studio suggested. Changing.